### PR TITLE
Add Record interface with CDC unwrap tx

### PR DESCRIPTION
**Pull request by jmar910**
Fixes https://github.com/meroxa/turbine-project/issues/215

## Description
This PR updates the interface for the  `RecordsArray` object that is passed in the data app's function, as well as the individual `Record` objects.

+ Added an `unwrapCDC` method to `RecordsArray`, allowing users to optionally unwrap CDC formatted data and send it along to destinations. This method will only operate on CDC formatted data, and will no op otherwise. As a result, this addresses MySQL -> fn -> PG data apps. 

+ Added `get` and `set` methods to the `Record`, which allows users to read and write data regardless of the underlying format (CDC or not, as long as it is a JSON object)

## Function
This updates the function interface for users and significantly reduces the amount of boilerplate users have to use in their functions

BEFORE
```js
  anonymize(records) {
    records.forEach((record) => {
      let payload = record.value.payload;
      if (
        isAttributePresent(payload.after) &&
        isAttributePresent(payload.after.customer_email)
      ) {
        payload.after.customer_email = iAmHelping(
          stringHash(payload.after.customer_email).toString()
        );
      }
    });

    return records;
  }
```
  
AFTER
```js
  anonymize(records) {
    records.forEach((record) => {
      record.set(
        "customer_email",
        stringHash(record.get("customer_email")).toString()
      );
    });
    
    // Used for most destinations
    records.unwrapCDC();

    return records;
  }
 ```

### CODE REVIEW

File: templates/javascript/index.js [hunk 1]
```diff
@@ -1,35 +1,21 @@
...
-      // Remember to reflect changes in the `app.json` configuration
-      if (
-        isAttributePresent(payload.after) &&
-        isAttributePresent(payload.after.customer_email)
-      ) {
-        payload.after.customer_email = iAmHelping(
-          stringHash(payload.after.customer_email).toString()
-        );
-      }
+      // Use record `get` and `set` to read and write to your data
```
Comment (jayjayjpg) [inline templates/javascript/index.js:8]: It's super neat how much more simple the initial app template looks like!

File: templates/javascript/index.js [hunk 2]
```diff
@@ -1,35 +1,21 @@
...
+      record.set(
+        "customer_email",
+        stringHash(record.get("customer_email")).toString()
```
Comment (jayjayjpg) [inline templates/javascript/index.js:11]: What do you think of still including the `iAmHelping` function in the demo example to better much our examples in the docs and the turbine library READMEs?

**Review by jayjayjpg — COMMENTED** [review]
This reads great and thank you for the swift iteration! Do you have the time available to also add one or two simple unit tests for the `unwrap` function itself or for the `Process` function when used together with a record that requires unwrapping?

Also don't forget to enjoy the long weekend, I just wanted to leave the comment here before I forget it on Tuesday!

File: test/unit/fn-records-array-test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import sinon from "sinon";
+import { RecordsArray } from "../../src/function-deploy/function-app/record";
+const CDCFixture = require("../../templates/javascript/fixtures/demo-cdc.json");
+
+QUnit.module("Unit | fn-records-array", () => {
```
Comment (jayjayjpg) [inline test/unit/fn-records-array-test.ts:5]: Neat! :sparkles:

**Review by jayjayjpg — APPROVED** [review]
This reads great, thank you for the super quick implementation of it! :sparkles:

File: README.md [hunk 4]
```diff
@@ -126,6 +118,55 @@ await destination.write(anonymized, "collection_archive");
...
+record.set('nested.key.down.some.levels', 'some_value')
+```
+
+
+The `records` parameter itself comes with an optional but important function
+
+```js
+records.unwrapCDC();
+```
+A user can optionally use this transform in their data app function to unwrap CDC formatted data into the right shape that most destinations expect. In instances where data is going to a destination, a user _will_ want call this (usually at the end of the data app function).
```
Comment (ericcheatham) [inline README.md:163]:
```suggestion
A user can optionally use this transform in their data app function to unwrap CDC formatted data into the right shape that most destinations expect. In instances where data is going to a destination, a user _will_ want to call this (usually at the end of the data app function).
```

**Comment by jayjayjpg** [comment]
Related to https://github.com/meroxa/turbine-project/issues/222

**Pull request merged**